torch/_inductor/fx_passes/decompose_mem_bound_mm.py

```diff
@@ -0,0 +1,123 @@
+from typing import Optional
+
+import torch
+from torch import Tensor
+from torch._dynamo.utils import counters
+from torch._inductor import utils
+
+from ..pattern_matcher import (
+    Arg,
+    CallFunction,
+    config_flag,
+    Match,
+    register_graph_pattern,
+)
+from .post_grad import decompose_mm_pass
+
+aten = torch.ops.aten
+
+# TODO: need a better strategy for decomposing mm
+MIN_FIRST_DIMENSION_DECOMPOSITION = 10240
+MAX_OTHER_DIMENSION_DECOMPOSITION = 32
+
+
+def check_device(a: Tensor, b: Tensor) -> bool:
+    return a.is_cuda and b.is_cuda
+
+
+def should_decompose_common(
```

Comment:
Can we check that the matmul is actual memory bound here ? 

https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/pad_mm.py#L180

Reply:
![image](https://github.com/pytorch/pytorch/assets/10678066/4cb85cd2-514c-4452-88de-a1644da9d4b0)
![image](https://github.com/pytorch/pytorch/assets/10678066/77207cda-0b88-4e54-862a-c2cd648fbe01)

I have tried to use 2 ways to calculate compute intensity. 1/(1/m+1/n+1/k), m*n*k/(m*n+n*k+m*k). However, the result shows that not all the memory bound mms can be decomposed.